src/c++/library/http_client.h

```diff
@@ -99,10 +143,13 @@ class InferenceServerHttpClient : public InferenceServerClient {
   /// base path in the following format: host:port/<base-path>.
   /// \param verbose If true generate verbose output when contacting
   /// the inference server.
+  /// \param ssl_options Specifies the settings for configuring
+  /// SSL encryption and authorization.
   /// \return Error object indicating success or failure.
   static Error Create(
       std::unique_ptr<InferenceServerHttpClient>* client,
-      const std::string& server_url, bool verbose = false);
+      const std::string& server_url, bool verbose = false,
+      const HttpSslOptions& ssl_options = HttpSslOptions());
```

Comment:
Since ssl_options is initialized by default constructor, won't it enable SSL based on default values for it?

Reply:
My question is do we want to have SSL enabled by default?

Reply:
I have added dcoumentation to describe how these options work. Essentially, they are SSL settings for client. Whether or not SSL is used depends upon the server which client is talking to. These options are ignored if server endpoint is not using SSL.  Basically, we need `https:// ` in the server url to use SSL.